File: csharp/0003无重复字符的最长子串.cs
public partial class Solution
{
    public int LengthOfLongestSubstring(string s)
    {
        Queue<char> que = new();
        int res = 1;
        if(s.Length>0)
        {
            que.Enqueue(s[0]);

            foreach (char item in s[1..])
            {
                while (que.Contains(item))
                {
                    que.Dequeue();
                }

                que.Enqueue(item);
                res = Math.Max(res, que.Count);
            }
        }

        else
            return 0;

        return res;
    }
}

File: python/main/0003无重复字符的最长子串.py

def lengthOfLongestSubstring(s: str) -> int:
    
    size=len(s)
    if(size==0):
        return 0
    res=1
    dp=1
    start=0

    for i in range(1,size):
        flag=False
        for j in range(i-1,start-1,-1):
            if(s[j]==s[i]):
                start=j+1
                dp=i-j
                flag=True
        
        if(not flag):
            dp+=1
        res=max(res,dp)
    return res

print(lengthOfLongestSubstring(""))
